src/components/common/Map.js

```diff
@@ -87,11 +94,29 @@ class Map extends React.Component {
   }
 
   onMapClick(e) {
-    let lat = parseFloat(e.latLng.lat().toFixed(6));
-    let lng = parseFloat(e.latLng.lng().toFixed(6));
-    this.props.updateTargetInfo("lat", lat);
-    this.props.updateTargetInfo("lng", lng);
-    this.props.updateTargetInfo("isVisible", true);
+    const lat = parseFloat(e.latLng.lat().toFixed(6));
+    const lng = parseFloat(e.latLng.lng().toFixed(6));
+    const { updateTargetInfo, unselectTarget, changeContent } = this.props;
+    unselectTarget();
+    updateTargetInfo("lat", lat);
```

Comment:
Ok, maybe i can update three fields in one call but i can't update the entire object, this is because targetInfo is not only assembled by lat, lng and isVisible in map component, but also some fields are filled in the targetForm Component.
To resume, if i update the whole object, the other fields will be ignored.